org.opent2t.sample.lamp.superpopular/com.wink.lightbulb/js/tests/test.js

```diff
@@ -51,4 +51,65 @@ test.serial('Power', t => {
                     });
                 });
         });
-});
\ No newline at end of file
+});
+
+// Get the entire Lamp schema object
+test.serial('GetLampResURI', t => {
+
+    return OpenT2T.createTranslatorAsync(translatorPath, 'thingTranslator', config.Device)
+        .then(translator => {
+            // TEST: translator is valid
+            t.is(typeof translator, 'object') && t.truthy(translator);
+
+            return OpenT2T.invokeMethodAsync(translator, 'org.opent2t.sample.lamp.superpopular', 'getLampResURI', []);
+        }).then((response) => {
+            t.not(response.id, undefined);
+            t.is(response.rt, 'org.opent2t.sample.lamp.superpopular');
+            t.not(response.power, undefined);
+            t.not(response.dim, undefined);
+
+            console.log('*** response: \n' + JSON.stringify(response, null, 2));
+        });
+});
+
+
+// Get the entire Lamp schema object
+test.serial('PostLampResURI_Set_Power', t => {
+
+    return OpenT2T.createTranslatorAsync(translatorPath, 'thingTranslator', config.Device)
+        .then(translator => {
+            // TEST: translator is valid
+            t.is(typeof translator, 'object') && t.truthy(translator);
+
+            var value = {};
+            value['power'] = { 'value': true };
+
+            return OpenT2T.invokeMethodAsync(translator, 'org.opent2t.sample.lamp.superpopular', 'postLampResURI', [value]);
+        }).then((response) => {
+            t.truthy(response.power.value);
+
+            console.log('*** response: \n' + JSON.stringify(response, null, 2));
+        });
+});
+
+// Set the name and dimming for the Lamp
+test.serial('PostLampResURI_Set_NameAndDim', t => {
+
+    return OpenT2T.createTranslatorAsync(translatorPath, 'thingTranslator', config.Device)
+        .then(translator => {
+            // TEST: translator is valid
+            t.is(typeof translator, 'object') && t.truthy(translator);
+
+            var value = {};
+            value['n'] = "opent2t light";

```

Comment:
It at least works for lights. Despite Wink's documentation showing that the 'name' field is writable on devices in general, when I tried this out on a Nest thermostat attached to Wink, the REST calls were failing. I noticed that the name isn't editable in their app either, so I don't know if that is just a specific limitation for the Nest thermostat or thermostats in general.

---
In reply to: [76939487](https://github.com/openT2T/translators/pull/87#discussion_r76939487) [](ancestors = 76939487)